### DISCUSSION TITLE
Linux Mint 22 / flet 0.25.0 / Linux Desktop App no longer works

### DISCUSSION THREAD

**Discussion by alexrjs** [flet-dev/flet#4454] [General]
My flet 0.24.2 app was successfully running as a desktop app under Linux Mint 22. I've upgraded to flet 0.25.0 first with desktop-light and then desktop. The app compiles fine with 

> flet build --clear-cache --org $(ORG) --company $(COMPANY) --copyright $(COPYRIGHT) --description $(DESCRIPTION) --project $(PROJECT) --build-version $(VERSION) $(OS) src/ | tee build.log
> [19:52:59] Created Flutter bootstrap project from gh:flet-dev/flet-build-template with ref 0.25.0 ✅      
>            Customized app icons and splash images ✅                                                      
> [19:53:09] Generated app icons ✅                                                                         
> [19:53:31] Packaged Python app ✅                                                                         
> [19:54:42] Built app for Linux ✅                                                                         
>            Copied build to build/linux directory ✅                                                       
>            Successfully built your app for Linux! 🥳 Find it in build/linux directory. 📁   
>       
But when I start the desktop app, it doesn't show the main window anymore. Is there a way to figure out why? And btw. my app runs perfectly under VSCode in run and debug mode. Even the nasty Textfield fokus problem is gone with flet 0.25.0.
So how can I figure out, why the compiled desktop app doesn't show the main window any more?

I used 'pip install 'flet[all]' --upgrade' to upgrade and later 'pip uninstall flet-desktop-light --yes' and 'pip install flet-desktop', and 'pip list | grep flet' shows:
flet                  0.25.0
flet-cli              0.25.0
flet-core             0.24.1
flet-desktop          0.25.0
flet-runtime          0.24.1
flet-web              0.25.0

**Comment by ndonkoHenri** [comment]
Sorry for that!
Upgrade to the new flet v0.25.1 which has some bug-fixes.

**Reply by alexrjs** [reply]
Thanks for getting back to me so quick. Still not working, unfortunately.

Things I did:
1.) removed all flet modules from the virtualenvironment
2.) did the recommended pip install 
3.) pip list | grep flet
flet 0.25.1
flet-cli 0.25.1
flet-desktop-lite 0.25.1
flet-web 0.25.1
4.) removed build folder
5.) did the build as described above
6.) build was successful
7.) still no main window
8.) removed build folder again
9.) upgraded to flet-dekstop 0.25.1 the recommended way
10.) redid the build with this module, also no success
11.) upgraded the linux flutter to 3.24.5, before I had 3.24.1
12.) made a new build with flet-dekstop 0.25.1, unfortunately no success
13.) Switched to a simple one py file flet sample and did a build of this demo, also no success getting a Main Window shown. And my complex app and the simple demo app work fine with VSCode and flet 0.25.1. 

So unless I find a way to get output from the app reporting errors in some way, I'm stuck with flet 0.24.1 I guess.

Anyway thanks for flet and your hard work... And I guess, I still can try to make it a web app... ;-)

**Comment by alexrjs** [comment]
So I applied some ChatGPT power and modified the my_application.cc in the build/flutter/linux folder.

```
#include "my_application.h"

#include <flutter_linux/flutter_linux.h>
#ifdef GDK_WINDOWING_X11
#include <gdk/gdkx.h>
#endif

#include "flutter/generated_plugin_registrant.h"
#include <stdio.h> // Include for debugging output

struct _MyApplication {
  GtkApplication parent_instance;
  char** dart_entrypoint_arguments;
};

G_DEFINE_TYPE(MyApplication, my_application, GTK_TYPE_APPLICATION)

// Implements GApplication::activate.
static void my_application_activate(GApplication* application) {
  MyApplication* self = MY_APPLICATION(application);

  printf("[DEBUG] Activating application...\n");

  GtkWindow* window = GTK_WINDOW(gtk_application_window_new(GTK_APPLICATION(application)));
  gtk_window_set_title(window, "Flutter Application");
  gtk_window_set_default_size(window, 1280, 720);

  printf("[DEBUG] Window created with title 'Flutter Application'.\n");

  FlDartProject* project = fl_dart_project_new();
  fl_dart_project_set_dart_entrypoint_arguments(project, self->dart_entrypoint_arguments);

  FlView* view = fl_view_new(project);
  gtk_widget_show(GTK_WIDGET(view));
  gtk_container_add(GTK_CONTAINER(window), GTK_WIDGET(view));

  printf("[DEBUG] Flutter view added to the GTK window.\n");

  gtk_widget_show(GTK_WIDGET(window));

  printf("[DEBUG] GTK window displayed.\n");
}

// Implements GApplication::startup.
static void my_application_startup(GApplication* application) {
  MyApplication* self = MY_APPLICATION(application);

  printf("[DEBUG] Starting application...\n");

  G_APPLICATION_CLASS(my_application_parent_class)->startup(application);

  printf("[DEBUG] Parent class startup completed.\n");

  self->dart_entrypoint_arguments = NULL;
}

static void my_application_dispose(GObject* object) {
  MyApplication* self = MY_APPLICATION(object);

  printf("[DEBUG] Disposing application instance...\n");

  g_clear_pointer(&self->dart_entrypoint_arguments, g_strfreev);

  G_OBJECT_CLASS(my_application_parent_class)->dispose(object);

  printf("[DEBUG] Application instance disposed.\n");
}

static void my_application_class_init(MyApplicationClass* klass) {
  G_APPLICATION_CLASS(klass)->activate = my_application_activate;
  G_APPLICATION_CLASS(klass)->startup = my_application_startup;
  G_OBJECT_CLASS(klass)->dispose = my_application_dispose;

  printf("[DEBUG] Application class initialized.\n");
}

static void my_application_init(MyApplication* self) {
  printf("[DEBUG] Application instance initialized.\n");
}

MyApplication* my_application_new() {
  printf("[DEBUG] Creating new application instance.\n");
  return MY_APPLICATION(g_object_new(my_application_get_type(), "application-id", "com.example.flutter", NULL));
}
```

Which generates the following output:

> [DEBUG] Creating new application instance.
> [DEBUG] Application class initialized.
> [DEBUG] Application instance initialized.
> [DEBUG] Starting application...
> [DEBUG] Parent class startup completed.
> [DEBUG] Activating application...
> [DEBUG] Window created with title 'Flutter Application'.
> [DEBUG] Flutter view added to the GTK window.
> [DEBUG] GTK window displayed.
> [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MissingPluginException(No implementation found for method ensureInitialized on channel window_manager)
> #0      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:332)
> <asynchronous suspension>
> #1      WindowManager.ensureInitialized (package:window_manager/src/window_manager.dart:104)
> <asynchronous suspension>
> #2      setupDesktop (package:demo/main.dart:356)
> <asynchronous suspension>
> #3      main (package:demo/main.dart:99)
> <asynchronous suspension>
> 
> [DEBUG] Disposing application instance...
> [DEBUG] Application instance disposed.

Interestingly enough this modified code brings up an empty window, while the original code does not produce any window at all.

But now I'm stuck, since the following error doesn't mean anything to me:
**_`[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MissingPluginException(No implementation found for method ensureInitialized on channel window_manager)`_**

Any help on that?

**Reply by ndonkoHenri** [reply]
Does this happen with a helloworld app?

**Reply by alexrjs** [reply]
I used the sample provided at the ElevateButton:
```
import flet as ft

def main(page: ft.Page):
    page.title = "Basic elevated buttons"
    page.add(
        ft.ElevatedButton(text="Elevated button"),
        ft.Button("Disabled button", disabled=True),
    )

ft.app(main)
```
Put it into a "main.py" and used the following command to generate the linux executable: 

> flet build --project demo linux .

And after the first compile I changed the "my_application.cc" code with the chatgpt code during build, so it is not reset. The build is successful and when I execute the demo executable, it delivers above output. 

Hope that helps...

**Reply by alexrjs** [reply]
@ndonkoHenri - Tried the above demo code (in main.py) with flet 0.25.2, still standalone linux app (built with "flet build --project demo linux .") does not work. Application starts up without error, but no output or window. Ctrl-C needed to close app. 

VSCode run and debug work fine with flet 0.25.2.

Updated as usual with "pip install 'flet[all]' --upgrade".

```
pip list | grep flet lists:
flet                  0.25.2
flet-cli              0.25.2
flet-desktop-light    0.25.2
flet-web              0.25.2
```

```
flet build --project demo linux .
[16:06:19] Created Flutter bootstrap project from gh:flet-dev/flet-build-template with ref 0.25.2 ✅      
[16:06:20] Customized app icons and splash images ✅                                                      
[16:06:31] Generated app icons ✅                                                                         
[16:06:51] Packaged Python app ✅                                                                         
[16:08:46] Built app for Linux ✅                                                                         
           Copied build to build/linux directory ✅                                                       
           Successfully built your app for Linux! 🥳 Find it in build/linux directory. 📁    
```
           

```
lsb_release -a
No LSB modules are available.
Distributor ID: Linuxmint
Description:    Linux Mint 22
Release:        22
Codename:       wilma
```

`cat main.py
`
```
import flet as ft

def main(page: ft.Page):
    page.title = "Basic elevated buttons"
    page.add(
        ft.ElevatedButton(text="Elevated button"),
        ft.Button("Disabled button", disabled=True),
    )

ft.app(main)
```

**Comment by mfxbe** [comment]
Can't help in any way with a solution, but this probably has the same background as my issue #4520.

**Comment by alexrjs** [comment]
Problem no longer exists in 0.26.0. 
Working Software Stack:
Linux Mint 22.1

Python 3.12.3

Flutter 3.27.3 • channel stable • https://github.com/flutter/flutter.git
Framework • revision c519ee916e (11 days ago) • 2025-01-21 10:32:23 -0800
Engine • revision e672b006cb
Tools • Dart 3.6.1 • DevTools 2.40.2

flet-cli              0.26.0
flet-desktop-light    0.26.0
flet-web              0.26.0

Hardware: Apple Macbook Pro 2014

But there are now some errors when running. Don't know if they are relevant, since they do not effect my flet app. Maybe they are caused by my HW (old mac), SW (Linux) combination!?

embedder.cc (2439): 'FlutterEngineRemoveView' returned 'kInvalidArguments'. Remove view info was invalid. The implicit view cannot be removed.

(demo:10463): GLib-GObject-CRITICAL **: 20:55:20.947: g_object_weak_unref: couldn't find weak ref 0x7114782cc820(0x5582aa9c37c8)
Segmentation fault (core dumped)
